internal/wasm/func_validation.go

```diff
@@ -904,28 +928,64 @@ func (s *valueTypeStack) popStackLimit() {
 	}
 }
 
-func (s *valueTypeStack) pushStackLimit() {
-	s.stackLimits = append(s.stackLimits, len(s.stack))
+// pushStackLimit pushes the control frame's bottom of the stack.
+func (s *valueTypeStack) pushStackLimit(params int) {
```

Comment:
`limit` is ambiguous so you can rename it as you like e.g.
```suggestion
func (s *valueTypeStack) pushStackFloor(params int) {
```